[browser][tests][outerloop] Standup OuterLoop tests on CI

kjpou1 opened the pull request:
Get the outerloop CI tests working.

To run the outerloop tests from PR:
```
/azp run runtime-libraries-mono outerloop
```

kjpou1 committed:
[browser][tests][outerloop] Standup OuterLoop tests on CI

kjpou1 committed:
System.Collections.NonGeneric.Tests passing

kjpou1 committed:
System.Console.Tests passing

kjpou1 committed:
System.Diagnostics.DiagnosticSource.Tests passing

MaximLipnin commented on src/libraries/System.Collections.Concurrent/tests/BlockingCollectionTests.cs:
```diff
@@ -744,6 +744,7 @@ public static void TestAddAnyTakeAny_Longrunning(int numOfAdds, int numOfTakes,
         [Theory]
         [InlineData(4, 2048, 2, 64)]
         [OuterLoop]
+        [PlatformSpecific(~TestPlatforms.Browser)] // Cannot wait on monitors on this runtime.
```
AFAIR we used to add `PlatformDetection.IsThreadingSupported` for case ` Cannot wait on monitors on this runtime.`:

`[ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]`.

Though, not sure if it's still so.

kjpou1 replied:
Thanks Maxim

kjpou1 replied:
These should all be modified now to use `[ConditionalXXX(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]`.

Thanks for the heads up on this @MaximLipnin

steveharter replied:
Does `IsThreadingSupported` capture the meaning? Or is it more about locking? We have lots more cases of "threading" than the ones here in this PR which do not have this conditional attribute.

kjpou1 committed:
System.Globalization.Tests passing

- See active issue https://github.com/dotnet/runtime/issues/45951

kjpou1 committed:
System.IO.FileSystem.Tests passing after deactivating test cannot wait on monitors on this runtime.

kjpou1 committed:
System.IO.FileSystem.Tests passing - See active issue: https://github.com/dotnet/runtime/issues/45954

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Remove fat fingered code.

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-tests-outerloop

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Threads not supported on Browser

kjpou1 committed:
Browser does not support monitor

kjpou1 committed:
Browser does not support monitor

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Browser does not support wait on monitors

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-tests-outerloop

kjpou1 committed:
Browser does not support wait on monitors

kjpou1 committed:
Add ActiveIssue for DataFlow tests TestNullTarget_CompletionNoCaching

kjpou1 committed:
Add ActiveIssue for DataFlow tests

kjpou1 committed:
Add ActiveIssue for DataFlow tests

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Add threading support conditional

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Change all Cannot wait on monitors on this runtime Browser platform checks to

[ConditionalXXXX(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]

As was suggested in comments

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

radical commented on src/libraries/Common/tests/System/Net/WebSockets/WebSocketCreateTest.cs:
```diff
@@ -40,6 +40,7 @@ public void CreateFromStream_ValidBufferSizes_CreatesWebSocket()
         [OuterLoop("Uses external servers")]
         [Theory]
         [MemberData(nameof(EchoServers))]
+        [PlatformSpecific(~TestPlatforms.Browser)] // System.Net.Sockets is not supported on this platform.
```
Should this be converted  to a new `PlatformDetection`, since there are other tests disabled for the same reason too?

kjpou1 replied:
Should this really be in this PR?  Maybe another PR?

radical replied:
Sure

radical commented on src/libraries/System.Diagnostics.DiagnosticSource/tests/DiagnosticSourceTests.cs:
```diff
@@ -506,6 +506,7 @@ public void AllListenersCheckCatchupList()
         [InlineData(100, 102)]
         [InlineData(100, 103)]
         [InlineData(100, 104)]
+        [PlatformSpecific(~TestPlatforms.Browser)] // Cannot wait on monitors on this runtime.
```
Same `PlatformDetection.IsThreadingSupported` change here, and other tests which are disabled because of monitors.

kjpou1 replied:
Addressed in recent commits

radical commented on src/libraries/System.Threading/tests/ThreadLocalTests.cs:
```diff
@@ -369,6 +369,7 @@ public static void RunThreadLocalTest9_Uninitialized()
 
         [Fact]
         [OuterLoop]
+        [PlatformSpecific(~TestPlatforms.Browser)] // Operation is not supported on this platform.
```
Just curious, what operation is not supported here?

kjpou1 replied:
[FAIL] System.Threading.Tests.ThreadLocalTests.ValuesGetterDoesNotThrowUnexpectedExceptionWhenDisposed
  info: System.PlatformNotSupportedException : Operation is not supported on this platform.
  info:    at System.Threading.Thread.Start()
  info:    at System.Threading.Tests.ThreadLocalTests.ValuesGetterDoesNotThrowUnexpectedExceptionWhenDisposed()
  info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo cultu

akoeplinger replied:
should be `[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]` then though?

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-tests-outerloop

kjpou1 committed:
Address review comment for monitors not supported

radical commented on src/libraries/System.Transactions.Local/tests/CloneTxTests.cs:
```diff
@@ -165,6 +165,7 @@ public void Run(CloneType cloneType, IsolationLevel isoLevel, bool forcePromote,
         [InlineData(CloneType.BlockingDependent, IsolationLevel.Snapshot, false, TransactionStatus.Aborted)]
         [InlineData(CloneType.BlockingDependent, IsolationLevel.Chaos, false, TransactionStatus.Aborted)]
         [InlineData(CloneType.BlockingDependent, IsolationLevel.Unspecified, false, TransactionStatus.Aborted)]
+        [PlatformSpecific(~TestPlatforms.Browser)] // Cannot wait on monitors on this runtime.
```
-> PlatformDetection.IsThreadingSupported

kjpou1 replied:
Addressed in recent commits

radical commented on src/libraries/System.Transactions.Local/tests/NonMsdtcPromoterTests.cs:
```diff
@@ -2151,6 +2151,7 @@ public void PSPENonMsdtcAbortingCloneNotCompleted(bool promote)
         [Theory]
         [InlineData(false)]
         [InlineData(true)]
+        [PlatformSpecific(~TestPlatforms.Browser)] // Cannot wait on monitors on this runtime.
```
-> PlatformDetection.IsThreadingSupported

kjpou1 replied:
Addressed in recent commits

radical commented on src/libraries/System.Transactions.Local/tests/NonMsdtcPromoterTests.cs:
```diff
@@ -2164,6 +2165,7 @@ public void PSPENonMsdtcBlockingCloneCompletedAfterCommit(bool promote)
         [Theory]
         [InlineData(false)]
         [InlineData(true)]
+        [PlatformSpecific(~TestPlatforms.Browser)] // Cannot wait on monitors on this runtime.
```
-> PlatformDetection.IsThreadingSupported

kjpou1 replied:
Addressed in recent commits

radical commented on src/libraries/System.Transactions.Local/tests/TransactionScopeTest.cs:
```diff
@@ -507,6 +507,7 @@ public void RMFail1()
 
         [Fact]
         [OuterLoop] // 30 second timeout
+        [PlatformSpecific(~TestPlatforms.Browser)] // Cannot wait on monitors on this runtime.
```
-> PlatformDetection.IsThreadingSupported

kjpou1 replied:
Addressed in recent commits

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-tests-outerloop

kjpou1 committed:
Address review comments

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-tests-outerloop

kjpou1 committed:
Add threading supported condition.

```
fail: [FAIL] System.Text.Json.Serialization.Tests.EnumConverterTests.VeryLargeAmountOfEnumsToSerialize
[08:10:56] info: System.PlatformNotSupportedException : Cannot wait on monitors on this runtime.
```

kjpou1 commented:
/azp run runtime-libraries-mono outerloop

kjpou1 committed:
Merge branch 'master' of https://github.com/dotnet/runtime into wasm-tests-outerloop
